### PR TITLE
add UCSBSubjects Table and team02 dependencies

### LINKED ISSUE #13: Table component for UCSBSubjects
Table component for UCSBSubjects

Using the `UCSBDatesTable` component as a model, create a similar component for UCSBSubjects.

# ACs

- [x] The table has columns for database fields (i.e. the attributes of the corresponding `@Entity` class
- [x] There are fixtures under `/frontend/src/fixtures` for this data type, similar to the other fixtures files.
- [x] There is a storybook entry similar to the one for `UCSBDatesTable`
- [x] 100% coverage with `npm run coverage`
- [x] 100% coverage with `npx stryker run`

### LINKED ISSUE #33: Encyclopedia Brown and the Case of the Missing Backend 🔎
Our program has no relevant Spring Boot backend for the frontend tickets.

### Acceptance Criteria

Tests first.

- [x] The file `CollegiateSubredditControllerTests.java` exists and is in the appropriate directory[^1] and package[^2] `edu.ucsb.cs156.example.controllers`.
- [x] The file `UCSBSubjectControllerTests.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.controllers`.

Then the controllers.

- [x] The file `CollegiateSubredditController.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.controllers`.
- [x] The file `UCSBSubjectController.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.controllers`.

Finally the entities[^3].

- [x] The file `CollegiateSubreddit.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.entities`.
- [x] The file `UCSBSubject.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.entities`.

Oh I forgot the repositories.

- [x] The file `CollegiateSubredditRepository.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.repositories`.
- [x] The file `UCSBSubjectRepository.java` exists and is in the appropriate directory and package `edu.ucsb.cs156.example.repositories`.

No need for the earthquakes epic. We're writing our own using Mongo 🥭.

[^1]: There doesn't seem to be a separate package for the controller tests and the controllers themselves despite the files living in different directories, so when I say _appropriate directory_ I really mean _I don't know what I'm talking about_. I thought Java packages mirror the structure of the project.
[^2]: I don't know how to rename the package, but water under the bridge. `team03` is already in motion.
[^3]: No tests for the entities, because they're structs?

### PR DESCRIPTION
**Pull request by rsdharmaji**
add UCSBSubjects table and team02 files.

### CODE REVIEW

**Comment by jacksoncooper** [comment]
Not sure why our storybook service failed.
> Failed to connect to github.com port 443: Connection timed out

Frontend coverge (`npm run coverage`) is failing for functions and statements because we lack 100 percent coverage.

File: frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    const deleteMutation = useBackendMutation(
+        cellToAxiosParamsDelete,
+        { onSuccess: onDeleteSuccess },
+        ["/api/UCSBSubjects/all"]
+    );
+    // Stryker enable all 
+
+    // Stryker disable next-line all : TODO try to make a good test for this
+    const deleteCallback = async (cell) => { deleteMutation.mutate(cell); }
```
Comment (jacksoncooper) [inline frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js:27]: We acknowledge the lack of coverage for this line. We will come back to it when we have some sort of understanding of what's going on.

**Comment by rsdharmaji** [comment]
the code coverage issue might be out of our control?
`    // Stryker disable next-line all : TODO try to make a good test for this
    const deleteCallback = async (cell) => { deleteMutation.mutate(cell); }`
prof. Conrad wrote the TODO in lecture on 2/14, so don't know if we need to do this.

File: frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js (hunk 1 above)
Comment (jacksoncooper) [inline frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js:27]: We acknowledge the lack of coverage for this line. We will come back to it when we have some sort of understanding of what's going on.

File: frontend/src/tests/utils/UCSBSubjectUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    //         const result = editCallback(cell);
+
+    //         // assert
+
+    //         expect(mockToast).toHaveBeenCalledWith(expectedToast);
+           
+    //     });
+
+    // });
+});
```
Comment (jacksoncooper) [inline frontend/src/tests/utils/UCSBSubjectUtils.test.js:70]: Are we able to remove this?

**Review by jacksoncooper — COMMENTED** [review]
Annotations.

**Comment by jacksoncooper** [comment]
Looks good, will come back to coverage.

**Pull request merged**